File: 2091-Removing-Minimum-And-Maximum-Of-Array/Solution.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int minimumDeletions(vector<int>& nums) {
        int n = nums.size();
        if (n <= 2) return n;

        int minIdx = 0, maxIdx = 0;
        for (int k = 0; k < n; ++k) {
            if (nums[k] < nums[minIdx]) minIdx = k;
            if (nums[k] > nums[maxIdx]) maxIdx = k;
        }

        // Order the indices such that i is the leftmost and j is the rightmost
        int i = min(minIdx, maxIdx);
        int j = max(minIdx, maxIdx);

        // Option 1: Remove both elements from the front
        int removeFromFront = j + 1;

        // Option 2: Remove both elements from the back
        int removeFromBack = n - i;

        // Option 3: Remove the leftmost from the front and the rightmost from the back
        int removeFromBoth = (i + 1) + (n - j);

        return min({removeFromFront, removeFromBack, removeFromBoth});
    }
};

File: 2091-Removing-Minimum-And-Maximum-Of-Array/Solution.py
class Solution:
    def minimumDeletions(self, nums: list[int]) -> int:
        n = len(nums)
        if n <= 2:
            return n

        min_idx = nums.index(min(nums))
        max_idx = nums.index(max(nums))

        # Order the indices such that i is the leftmost and j is the rightmost
        i = min(min_idx, max_idx)
        j = max(min_idx, max_idx)

        # Evaluate the 3 possible deletion strategies
        return min(
            j + 1,             # Remove both from front
            n - i,             # Remove both from back
            (i + 1) + (n - j)  # Remove from both ends
        )
      
